dev/core#5877 Fix loss of tabs

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
dev/core#5877 Fix loss of tabs

Before
----------------------------------------
![image](https://github.com/user-attachments/assets/abc25405-283c-4fa2-9bae-e1145b3e5fb1)


After
----------------------------------------
![image](https://github.com/user-attachments/assets/792581fe-5fb7-480e-9027-d98d4e4f82a9)


Technical Details
----------------------------------------

Comments
----------------------------------------

eileenmcnaughton committed:
dev/core#5877 Fix loss of tabs

demeritcowboy commented on CRM/Contact/Page/View/Summary.php:
```diff
@@ -242,7 +234,11 @@ public function view() {
-    $this->assign('allTabs', $this->getTabs($defaults));
+    $this->assign('allTabs', $this->getTabs([
+      'contact_type' => $defaults['contact_type'],
+      'contact_sub_type' => $defaults['contact_sub_type'],
```
I'm getting an error about undefined contact_sub_type here (the site has no such things).

Otherwise though looks much better.

eileenmcnaughton replied:
Ah yes - cos that function helpfully filters out empty fields - I've switched to our more recent apiv4 based methodology - it does do an extra look up but we can use it remove some others

eileenmcnaughton committed:
Add getContactID public function to ContactSummary page

eileenmcnaughton committed:
Use form helper to get contact details

This does do an extra lookup at this stage - but the trade off is we reliably
have cached apiv4 style values available here, consistent with other forms
